### PR TITLE
Add SourceFile utility and EmbedSourceModifier for mdoc

### PR DESCRIPTION
**Pull request by khajavi**
## Summary

Promote source file embedding utilities from zio-http into zio-sbt-source as reusable library.

- **SourceFile** — `readSource`, `fileExtension`, `printSource` for embedding source files in mdoc docs
  - Handles relative paths with `../` fallback (mdoc runs in docs subdir)
  - Proper resource cleanup via try/finally
  - Both Scala 2.13 + 3.3.8 supported

- **EmbedSourceModifier** — mdoc PostModifier wrapping SourceFile
  - Usage: ` ```scala mdoc:embed:path/to/file.scala ` ` (no imports needed)
  - Auto-discovered via ServiceLoader (META-INF/services registration)
  - mdoc 2.5.4 added as dependency

## Test Coverage

- 30 tests pass on Scala 2.13 and 3.3.8
- fileExtension edge cases (no ext, mixed cases)
- readSource full file + single-line
- printSource formatting with/without title, with/without line numbers
- Temp file isolation using ZIO Test patterns

## Usage

In any mdoc documentation:

\`\`\`scala mdoc:embed:zio-http-example/src/main/scala/example/HelloWorld.scala
\`\`\`

Projects add \`zio-sbt-source\` as dependency; modifier is auto-discovered.

🤖 Generated with Claude Code

### CODE REVIEW

File: zio-sbt-source/src/main/scala/zio/sbt/SourceFile.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+object SourceFile {
+
+  def readSource(path: String): String = {
+    val f          = new JFile(path)
+    val actualPath = if (f.exists()) path else "../" + path
+    val src        = Source.fromFile(actualPath)
+    try src.getLines().mkString("\n")
+    finally src.close()
+  }
```
[the remark this answers is not in the document]
Comment (khajavi) [reply zio-sbt-source/src/main/scala/zio/sbt/SourceFile.scala:33]: Fixed: Added explicit UTF-8 charset to `Source.fromFile()` call in `readSource()`. Now matches `SourceReader` behavior and avoids platform-dependent charset issues.

File: zio-sbt-source/src/main/scala/zio/sbt/SourceFile.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  def fileExtension(path: String): String =
+    Paths.get(path).getFileName.toString.split('.').lastOption.getOrElse("")
```
[the remark this answers is not in the document]
Comment (khajavi) [reply zio-sbt-source/src/main/scala/zio/sbt/SourceFile.scala]: Fixed: Added null check for `Paths.get().getFileName()`. Returns empty string for empty or root paths instead of throwing NPE.

File: zio-sbt-source/src/main/scala/zio/sbt/EmbedSourceModifier.scala [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  override val name = "embed"
+
+  override def process(ctx: PostModifierContext): String = {
+    val parts = ctx.info.split(":")
+    val path  = if (parts.length > 1) parts(1) else ""
+
+    if (path.isEmpty) {
+      ctx.reporter.error(
+        s"embed modifier requires a path argument: ```scala mdoc:embed:path/to/file.scala"
+      )
```
[the remark this answers is not in the document]
Comment (khajavi) [reply zio-sbt-source/src/main/scala/zio/sbt/EmbedSourceModifier.scala:33]: Fixed: Removed unclosed code fence from error message. Now uses inline text "mdoc:embed:path/to/file.scala" without fenced block.

File: zio-sbt-source/src/main/scala/zio/sbt/EmbedSourceModifier.scala [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import mdoc.PostModifier
+import mdoc.PostModifierContext
+
+class EmbedSourceModifier extends PostModifier {
+  override val name = "embed"
+
+  override def process(ctx: PostModifierContext): String = {
+    val parts = ctx.info.split(":")
+    val path  = if (parts.length > 1) parts(1) else ""
+
```
[the remark this answers is not in the document]
Comment (khajavi) [reply zio-sbt-source/src/main/scala/zio/sbt/EmbedSourceModifier.scala]: Fixed: Replaced fragile split-based parsing with `stripPrefix()` chain. Now correctly handles colons in paths (e.g., Windows paths, URIs).

File: zio-sbt-source/src/test/scala/zio/sbt/SourceFileSpec.scala [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+import zio.Scope
+import zio.test._
+
+object SourceFileSpec extends ZIOSpecDefault {
+
+  // Evaluate result eagerly before finally-delete runs; mirrors SourceReaderSpec pattern.
+  private def withTempFile[A](content: String)(f: String => A): A = {
+    val tmp = Files.createTempFile("source_file_spec_", ".scala")
+    try {
+      Files.write(tmp, content.getBytes("UTF-8"))
```
[the remark this answers is not in the document]
Comment (khajavi) [reply zio-sbt-source/src/test/scala/zio/sbt/SourceFileSpec.scala]: Fixed: Updated test to use `StandardCharsets.UTF_8` instead of string literal "UTF-8". Also updated EmbedSourceModifier to use StandardCharsets consistently.

**Comment by khajavi** [comment]
All review feedback has been addressed:
- ✅ UTF-8 charset handling (readSource)
- ✅ fileExtension null check (empty path handling)
- ✅ Error message formatting (EmbedSourceModifier)
- ✅ Robust path parsing (stripPrefix instead of split)
- ✅ StandardCharsets.UTF_8 usage consistency

Tests pass. Ready for re-review.
